### PR TITLE
INTMDB-648: Errors when creating or importing timeseries online archive

### PR DESCRIPTION
**Pull request by martinstibbe**
## Description

Errors when creating or importing timeseries online archive. Add support for collection_type

Link to any related issue(s):
https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1081
## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: mongodbatlas/resource_mongodbatlas_online_archive.go [hunk 1]
```diff
@@ -339,8 +346,10 @@ func resourceMongoDBAtlasOnlineArchiveUpdate(ctx context.Context, d *schema.Reso
 	paused := d.HasChange("paused")
 	criteria := d.HasChange("criteria")
 
+	collectionType := d.HasChange("collection_type")
+
 	// nothing to do, let's go
-	if !paused && !criteria {
+	if !paused && !criteria && collectionType {
```
Comment (evertsd) [inline mongodbatlas/resource_mongodbatlas_online_archive.go]: Should this be `&& !collectionType`?

Comment (martinstibbe) [reply mongodbatlas/resource_mongodbatlas_online_archive.go]: Good  catch updated

**Review by evertsd — APPROVED** [review]

File: mongodbatlas/data_source_mongodbatlas_online_archive.go [hunk 2]
```diff
@@ -83,6 +83,10 @@ func schemaOnlineArchive() map[string]*schema.Schema {
 			Type:     schema.TypeString,
 			Computed: true,
 		},
+		"collection_type": {
```
Comment (Zuhairahmed) [inline mongodbatlas/data_source_mongodbatlas_online_archive.go:86]: are we missing similar updates to data source mongodbatlas online archive**s**? https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/data-sources/online_archives

File: mongodbatlas/data_source_mongodbatlas_online_archive.go [hunk 3]
```diff
@@ -83,6 +83,10 @@ func schemaOnlineArchive() map[string]*schema.Schema {
...
+			Type:     schema.TypeString,
+			Computed: true,
```
Comment (Zuhairahmed) [inline mongodbatlas/data_source_mongodbatlas_online_archive.go:88]: also looks like we are missing _test.go file, i.e. data_source_mongodbatlas_online_archive_test.go. if true can you add it as part of this PR?

File: website/docs/d/online_archive.html.markdown [hunk 4]
```diff
@@ -33,6 +33,7 @@ data "mongodbatlas_online_archive" "test" {
 ## Attributes reference
 * `db_name`          -  Name of the database that contains the collection.
 * `coll_name`        -  Name of the collection.
+* `collection_name`  -  Classification of MongoDB database collection that you want to return
```
Comment (Zuhairahmed) [inline website/docs/d/online_archive.html.markdown]:
```suggestion
* `collection_name`  -  Classification of MongoDB database collection that you want to return, "TIMESERIES" or "STANDARD". Default is "STANDARD". 
```

File: website/docs/d/online_archives.html.markdown [hunk 5]
```diff
@@ -42,6 +42,7 @@ Each object in the results array represents an online archive with the following
 * `archive_id`         - ID of the online archive.
 * `db_name`          -  Name of the database that contains the collection.
 * `coll_name`        -  Name of the collection.
+* `collection_name`  -  Classification of MongoDB database collection that you want to return
```
Comment (Zuhairahmed) [inline website/docs/d/online_archives.html.markdown]:
```suggestion
* `collection_name`  -  Classification of MongoDB database collection that you want to return, "TIMESERIES" or "STANDARD". Default is "STANDARD". 
```

File: website/docs/r/online_archive.html.markdown [hunk 6]
```diff
@@ -76,6 +76,7 @@ resource "mongodbatlas_online_archive" "test" {
 * `cluster_name`     -  (Required) Name of the cluster that contains the collection.
 * `db_name`          -  (Required) Name of the database that contains the collection.
 * `coll_name`        -  (Required) Name of the collection.
+* `collection_name`  -  Classification of MongoDB database collection that you want to return
```
Comment (Zuhairahmed) [inline website/docs/r/online_archive.html.markdown]:
```suggestion
* `collection_name`  -  Classification of MongoDB database collection that you want to return, "TIMESERIES" or "STANDARD". Default is "STANDARD". 
```

**Review by Zuhairahmed — APPROVED** [review]
Few minor comments below and double check on potentially missing test harness / data source for plural. otherwise LGTM. feel free to merge after below edits

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: mongodbatlas/data_source_mongodbatlas_online_archive.go (hunk 2 above)
Comment (martinstibbe) [reply mongodbatlas/data_source_mongodbatlas_online_archive.go:86]: plural and singular all in same file will refactor in upcoming release

File: mongodbatlas/data_source_mongodbatlas_online_archive.go (hunk 3 above)
Comment (martinstibbe) [reply mongodbatlas/data_source_mongodbatlas_online_archive.go:88]: same resource has data source tests combined will fix in later release

**Pull request merged**